src/main/scala/org/broadinstitute/dsde/rawls/model/WorkspaceModel.scala

```diff
@@ -16,7 +21,24 @@ trait Attributable {
 
 trait DomainObject {
   //the name of a field on this object that uniquely identifies it relative to any graph siblings
-  def idField: String
+  protected def idField: Option[String]
+
+  // a function to uniquely identify this object relative to any graph siblings
+  // by default, it checks the idField for equality
+  def idFilterFn(tpe: Type)(vertex: Vertex): Boolean = {
+    this.idField match {
+      case Some(id) => getDomainObjectIdValue(tpe, this) == vertex.getProperty(id)
```

Comment:
I have serious reservations about this.

1. I don't like that our serialization code is leaking out of GraphDAO classes.
2. Not only is the serialization code leaking, but the specifics of *how* fields are serialized are leaking. Even if the answer to #1 is "tough crap", I think this should be a call to `GraphDAO.loadProperty()` in order to keep the implementation details encapsulated.